src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1330,6 +1342,32 @@ citus_update_node(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ *
```

Comment:
needs comment: What does this function do? Why within tx only?

We tend to forget these quickly, so better to document :)

Reply:
Added the comments